docs/blog/2024-09-02-microservices-and-workflow.md

```diff
@@ -0,0 +1,124 @@
+2. **Register a Workflow Specification** (we call it a [`WfSpec` in LittleHorse](https://littlehorse.dev/docs/concepts/workflows)) which specifies what tasks to execute and when.
+3. **Run your workflow** so that the workflow engine can orchestrate the process to completion.
+
+![LittleHorse Architecture](../static/img/2024-08-28-lh-application.png)
+
+[Task Workers](https://littlehorse.dev/docs/developer-guide/task-worker-development) are where a workflow can interface with the outside world. Since a Task in a workflow results in the LittleHorse SDK calling a programming function/method of your choosing, Task Workers allow LittleHorse to integrate with any system. Task Workers can make database queries, call external API's, provision infrastructure on AWS, send push notifications to customer mobile apps, perform calculations, call an LLM API, and more.
+
+In LittleHorse, the `WfSpec` is [defined in code](https://littlehorse.dev/docs/developer-guide/wfspec-development) in a language of your choice. Because LittleHorse was written with developers in mind, our DSL's have all of the primitives that you'd expect in a programming language: variables, control flow, exception handling, child threads, interrupts, and awaiting for external events. This allows workflows to be:
+
+* Easy to reason about.
+* Tracked in version control.
+* Familiar and easy to learn.
+
+Once you tell LittleHorse to [run an instance of your `WfSpec`](https://littlehorse.dev/docs/developer-guide/grpc/running-workflows), LittleHorse will oversee the entire process until it completes. Failed tasks will be retried, every step will be journaled, and the state of your processes will be safely and durably persisted while waiting for external triggers.
+
+## Why Workflow?
+
+Microservice applications are distributed workflows. This presents operational challenges because there is no "leader" providing oversight in microservice processes. Thankfully, a developer-focused and horizontally-scalable workflow engine like LittleHorse can fill the "leader" role, thus providing oversight and reliability, and taming the complexity of your business processes.
+
+Additionally, using a workflow engine allows you to develop a set of _resuable_ and _modular_ tasks which can be easily dropped into any business workflow with a common API. Rather than accumulating tech debt, workflow engines allow you to accumulate a set of useful lego bricks.
+
+In most existing orginizations there's a long list of API calls required to simply _run_ a workflow.  Training engineers to use all of the new APIs while securely distributing access and permissions causes confusion and slow development cycles.  Workflow engines provide a single API and single system that allows anyone to securely manage, run, and operate complex workflows.
```

Comment:
There's a little misspelling here. It should be _organization_ :D

Reply:
resuable is also spelled wrong

Reply:
Great catch, both of you!